src/core/sprinkler.rs

```diff
@@ -0,0 +1,133 @@
+use crate::core::data::{BobKey, BobData, Node, NodeDisk, VDisk, BobError};
+use crate::core::link_manager::LinkManager;
+
+use tokio::prelude::*;
+use std::time::{ Duration};
+use std::sync::Arc;
+
+use futures::future::*;
+use futures::stream::*;
+
+
+
+
+
+#[derive(Clone)]
+pub struct Cluster {
+    pub vdisks: Vec<VDisk>
+
+}
+
+#[derive(Clone)]
+pub struct Sprinkler {
+    cluster: Cluster,
+    quorum: u8,
+    link_manager: Arc<LinkManager>
+}
+
+pub struct SprinklerError {
+    total_ops: u16,
+    ok_ops: u16,
+    quorum: u8
+
+}
+
+pub struct SprinklerResult {
+    total_ops: u16,
+    ok_ops: u16,
+    quorum: u8
+}
+
+impl Sprinkler {
+    pub fn new() -> Sprinkler {
+        let ex_cluster = Cluster {
```

Comment:
Hardcoded data. Is this a temporary and will be changed in future PR?

Reply:
definitely